src/tests/unittests/validation/ValidationTest.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ValidationTest.h"
+
+#include "nxt/nxt.h"
+
+namespace backend {
+    namespace null {
+        void Init(nxtProcTable* procs, nxtDevice* device);
+    }
+}
+
+ValidationTest::ValidationTest() {
+    nxtProcTable procs;
+    nxtDevice cDevice;
+    backend::null::Init(&procs, &cDevice);
+
+    nxtSetProcs(&procs);
+    device = nxt::Device::Acquire(cDevice);
+}
+
+ValidationTest::~ValidationTest() {
+    // We need to destroy NXT objects before setting the procs to null otherwise the nxt*Release
+    // will call a nullptr
+    device = nxt::Device();
+    nxtSetProcs(nullptr);
+}
+
+void ValidationTest::TearDown() {
+    for (auto& expectation : expectations) {
+        // Double the gtest asserts with prints of the debug name to help with troubleshooting
+
+        if (!expectation.gotStatus) {
+            std::cerr << "Didn't get a status for " << expectation.debugName << std::endl;
+        }
+        ASSERT_TRUE(expectation.gotStatus);
```

Comment:
I think this works:

`ASSERT_TRUE(expectation.gotStatus) << "Didn't get a status for " << expectation.debugName;`

and would be cleaner.